### DISCUSSION TITLE
how can you choose which fields to bring from a populated field?

### DISCUSSION THREAD

**Discussion by sawa-ko** [mikro-orm/mikro-orm#2965] [Q&A] [answered]
Something like the fields option when searching with the find method.

The problem is that when I get the data from the user reference, I also get the sensitive data that I don't want to be available there.

```typescript
 const relation = this._relationsRepository.create({ ... });
 await this._relationsRepository.persistAndFlush(relation);
 const user = await relation.from_user.load();
 user.field; // data
 user.password; // as well as the fields option converts a property to undefind.
```
```typescript
 @ManyToOne({ entity: () => UserEntity, wrappedReference: true })
 from_user: IdentifiedReference<UserEntity, 'id' | '_id'>;
```

**Comment by B4nan** [comment, marked as answer]
Instead of `reference.load()` you should be using populate hints via `em.populate()` where you have both `fields` and `populate` option. You can combine those, or even omit the `populate` hint and just use `fields`. But for properties like `password` you should be using either lazy or hidden options, those are better suited for such.

But its valid point we should have `fields` in the `ref.load()` options too, currently we have just the populate hint there.
